lazy_tensor_core/lazy_tensor_core/csrc/ts_backend/TsNode.cpp

```diff
@@ -79,5 +86,18 @@ lazy_tensors::Shape TsNode::TsGetOpShape(
   return *shape;
 }
 
+std::string TsNode::ToString() const {
+  std::stringstream ss;
+  ss << shape() << " " << op();
```

Comment:
The only thing that is backend dependent here is `shape`. Would it be better to leave `ToString` in the `Node` and have each backend Node to implement a `ShapeToString()`? 

The downside of this approach I suggested would be
1. Each backend might want to implement `ToString` differently
2. Passing string as value might be expensive

The upside is backend doesn't need to manually update its `ToString` method when there is a new change in the `Node` base class (For example `aten::shape` you want to add). You also don't need to do a Cast in `Node` to get the backend shape for hash now.

Reply:
I think Backend can just override ToString entirely if they want, we can skip `ShapeToString` in this case.

Reply:
if we can get away putting actual fields in the base class by overriding `ToString` we should probably do that